src/app/starredentries/starredentries.component.ts

```diff
@@ -22,51 +18,40 @@ export class StarredEntriesComponent extends Base implements OnInit {
   user: any;
   starGazersClicked = false;
   organizationStarGazersClicked = false;
-  selectedEntry: any;
-  constructor(private starringService: StarringService,
-              private userQuery: UserQuery,
-              private imageProviderService: ImageProviderService,
-              private providerService: ProviderService,
-              private starentryService: StarentryService,
-              private usersService: UsersService) {
-                super();
-               }
+  constructor(
+    private userQuery: UserQuery,
+    private imageProviderService: ImageProviderService,
+    private providerService: ProviderService,
+    private usersService: UsersService
+  ) {
+    super();
+  }
 
   ngOnInit() {
-    this.starentryService.starEntry$.pipe(takeUntil(this.ngUnsubscribe)).subscribe(
-      entry => {
-        this.selectedEntry = entry;
-      }
-    );
-    this.userQuery.user$.subscribe(user => this.user = user);
-    this.usersService.getStarredTools().subscribe(
-      starredTool => {
-        this.starredTools = starredTool.filter((entry: DockstoreTool) => entry.is_published);
-        this.starredTools.forEach(
-          tool => {
-            if (!tool.providerUrl) {
-              this.providerService.setUpProvider(tool);
-            }
-            if (!tool.imgProviderUrl) {
-              tool = this.imageProviderService.setUpImageProvider(tool);
-            }
-          });
+    this.userQuery.user$.subscribe(user => (this.user = user));
+    this.usersService.getStarredTools().subscribe(starredTool => {
+      this.starredTools = starredTool.filter((entry: DockstoreTool) => entry.is_published);
+      this.starredTools.forEach(tool => {
+        if (!tool.providerUrl) {
+          this.providerService.setUpProvider(tool);
+        }
+        if (!tool.imgProviderUrl) {
+          tool = this.imageProviderService.setUpImageProvider(tool);
+        }
       });
-    this.usersService.getStarredWorkflows().subscribe(
-      starredWorkflow => {
-        this.starredWorkflows = starredWorkflow.filter((entry: Workflow) => entry.is_published);
-        this.starredWorkflows.forEach(
-          workflow => {
-            if (!workflow.providerUrl) {
-              this.providerService.setUpProvider(workflow);
-            }
-        });
+    });
+    this.usersService.getStarredWorkflows().subscribe(starredWorkflow => {
+      this.starredWorkflows = starredWorkflow.filter((entry: Workflow) => entry.is_published);
+      this.starredWorkflows.forEach(workflow => {
+        if (!workflow.providerUrl) {
```

Comment:
You could use a filter instead. Don't know that it's better though. :)

Reply:
I didn't actually change this. Hide whitespace changes when reviewing to remove a thousand+ line changes :)

Reply:
Interesting, I actually don't see how providerUrl is every not null